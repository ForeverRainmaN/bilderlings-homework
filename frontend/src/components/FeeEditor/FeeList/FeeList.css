.bilderlings-homework-fee-list {
  padding-top: 20px;
}