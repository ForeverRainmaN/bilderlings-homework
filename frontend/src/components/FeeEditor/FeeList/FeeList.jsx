import React from 'react';
import Fee from "./Fee/Fee";
import PropTypes from 'prop-types';
import './FeeList.css';

export default function FeeList({feeList, removeFee}) {
  return (
      <div className="bilderlings-homework-currency-pair-list">
        {feeList.map(fee =>
            <Fee key={fee.id}
                 from={fee.from}
                 to={fee.to}
                 fee={fee.fee}
                 removeFee={() => removeFee(fee.id)}
            />
        )}
      </div>
  );
}

FeeList.propTypes = {
  FeeList: PropTypes.shape({
    id: PropTypes.number.isRequired,
    from: PropTypes.string.isRequired,
    to: PropTypes.string.isRequired,
    fee: PropTypes.number.isRequired
  })
};