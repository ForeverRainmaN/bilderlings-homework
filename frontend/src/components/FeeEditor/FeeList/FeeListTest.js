import React from 'react';
import TestRenderer from 'react-test-renderer';
import FeeList from './FeeList';

const feeListToRender = [
  {
    id: 1,
    from: "USD",
    to: "EUR",
    fee: 1.25
  },
  {
    id: 2,
    from: "EUR",
    to: "RUB",
    fee: 1.05
  },
  {
    id: 3,
    from: "RUB",
    to: "EUR",
    fee: 1.06
  }
];

it('tests if FeeList renders all fees that are passed via props correctly',
    () => {
      const feeList = TestRenderer.create(
          <FeeList FeeList={feeListToRender} removeFee={() => {}} />
      )
    });