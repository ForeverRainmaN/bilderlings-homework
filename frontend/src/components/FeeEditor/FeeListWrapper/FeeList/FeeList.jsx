import React from 'react';
import CurrencyPair from "./Fee/Fee";
import PropTypes from 'prop-types';
import './FeeList.css';

export default function FeeList({feeList, removeFee}) {
  return (
      <div className="bilderlings-homework-currency-pair-list">
        {feeList.map(currencyPair =>
            <CurrencyPair key={currencyPair.id}
                          from={currencyPair.from}
                          to={currencyPair.to}
                          fee={currencyPair.fee}
                          removeFee={() => removeFee(currencyPair.id)}
            />
        )}
      </div>
  );
}

FeeList.propTypes = {
  FeeList: PropTypes.shape({
    id: PropTypes.number.isRequired,
    from: PropTypes.string.isRequired,
    to: PropTypes.string.isRequired,
    fee: PropTypes.number.isRequired
  })
};