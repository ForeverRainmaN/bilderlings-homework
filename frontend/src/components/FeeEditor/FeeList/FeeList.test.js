import React from 'react';
import ShallowRenderer from 'react-test-renderer';
import {mount} from 'enzyme';
import FeeList from './FeeList';

const feeListToRender = [
  {
    id: 1,
    from: "USD",
    to: "EUR",
    fee: 1.25
  },
  {
    id: 2,
    from: "EUR",
    to: "RUB",
    fee: 1.05
  },
  {
    id: 3,
    from: "RUB",
    to: "EUR",
    fee: 1.06
  }
];

it('tests if FeeList renders all fees that are passed via props correctly',
    () => {
      const feeList = ShallowRenderer
      .create(<FeeList
          feeList={feeListToRender}
          removeFee={() => {
          }}
      />).toJSON();
      expect(feeList).toMatchSnapshot();

      const mountedFeeList = mount(<FeeList feeList={feeListToRender}
                                            removeFee={() => ({})}/>);
      const fee = mountedFeeList.find('.bilderlings-homework-curr-pair');
      expect(fee).toHaveLength(3);
    });