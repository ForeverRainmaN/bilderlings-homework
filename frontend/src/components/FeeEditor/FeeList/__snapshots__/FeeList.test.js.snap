// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests if FeeList renders all fees that are passed via props correctly 1`] = `
<div
  className="bilderlings-homework-currency-pair-list"
>
  <div
    className="bilderlings-homework-curr-pair"
  >
    <span>
      USD
    </span>
    <span>
      EUR
    </span>
    <span>
      1.25
    </span>
    <button
      className="bp3-button bp3-intent-danger"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      type="button"
    >
      <span
        className="bp3-button-text"
      >
        Remove
      </span>
    </button>
  </div>
  <div
    className="bilderlings-homework-curr-pair"
  >
    <span>
      EUR
    </span>
    <span>
      RUB
    </span>
    <span>
      1.05
    </span>
    <button
      className="bp3-button bp3-intent-danger"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      type="button"
    >
      <span
        className="bp3-button-text"
      >
        Remove
      </span>
    </button>
  </div>
  <div
    className="bilderlings-homework-curr-pair"
  >
    <span>
      RUB
    </span>
    <span>
      EUR
    </span>
    <span>
      1.06
    </span>
    <button
      className="bp3-button bp3-intent-danger"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      type="button"
    >
      <span
        className="bp3-button-text"
      >
        Remove
      </span>
    </button>
  </div>
</div>
`;
